overlord/hookstate/hookstate_test.go

```diff
@@ -347,13 +352,22 @@ func (s *hookManagerSuite) TestHookTaskEnforcesMaxWaitTime(c *C) {
 	s.state.Unlock()
 
 	// Force the snap command to hang
-	s.command = testutil.MockCommand(c, "snap", "while true; do sleep 1; done")
+	cmd := testutil.MockCommand(c, "snap", "while true; do sleep 1; done")
+	defer cmd.Restore()
 
+	pgrp := 0
 	// simulate a process that can not be killed
-	restore := hookstate.MockSyscallKill(func(int, syscall.Signal) error {
+	restore := hookstate.MockSyscallKill(func(pid int, _ syscall.Signal) error {
+		pgrp = pid
 		return nil
 	})
 	defer restore()
+	defer func() {
+		// do kill the processes
+		if pgrp != 0 {
+			syscall.Kill(pgrp, 9)
```

Comment:
fwiw, I copied the prod code, is not using the constant either. I can see to change it